marimo/_cli/convert/commands.py

```diff
@@ -0,0 +1,48 @@
+# Copyright 2024 Marimo. All rights reserved.
+from __future__ import annotations
+
+from pathlib import Path
+
+import click
+
+from marimo._cli.convert.ipynb import convert_from_ipynb
+from marimo._cli.convert.markdown import convert_from_md
+from marimo._cli.convert.utils import load_external_file
+
+
+@click.argument("filename", required=True)
+def convert(
+    filename: str,
+) -> None:
+    """Convert a Jupyter notebook or Markdown file to a marimo notebook.
+
+    The argument may be either a path to a local .ipynb/.md file,
+    or an .ipynb/.md file hosted on GitHub.
+
+    Example usage:
+
+        marimo convert your_nb.ipynb > your_nb.py
+
+    or
+
+        marimo convert your_nb.md > your_nb.py
```

Comment:
@dmadisetti : Are there restrictions on how the Python code should appear in the `markdown`?

Does the code fence need a special tag, for example? I noticed

\`\`\`{.python.marimo}
...
\`\`\`

in the test snapshots.

If there is a restriction, can you document it?

Reply:
Yeah! I think the block delimiters are actually up for more discussion

Currently, conversion will match a codeblock that contains `{.*marimo.*}` on the first line.

For valid syntax highlighting (I checked for out of the box vscode), the export tag is `{.python.marimo}`, but in my local quarto testing I've just been using `{marimo}`. Quarto denotes `{python}` as execution and plain `python` as a normal code block, so I thought I'd borrow from that.

Livebook is still on my mind, and I noticed they don't do much special in the raw document and just use a plain codefence:

````md
```elixr
1 = x # elixr is weird
```
````

but the extension of the file is `*.livemd` to hint that the notebook is runnable (Quarto does `.qmd`). They also support 2 languages (erlang and elixr), so keeping it simple is probably best for them.

It also matters on the extent of markdown integration. Does marimo stop at just export/ convert (I know you are strongly committed to pure python notebooks, and I think that makes sense), or could `marimo edit notebook.md` be on the tables? It's not that much of a push with this update. In that case, maybe a shorter string than `{.python.marimo}` makes sense, and this should just be `python`. In the markdown header export, I have it put a `marimo-version` field, so it could cue off that and keep markdown to spec.

Or! It can stay as `{.python.marimo}`, which seems like an alright compromise. Is there somewhere other than here to update https://docs.marimo.io/getting_started/index.html#convert-jupyter-notebooks ?

Reply:
> Quarto denotes `{python}` as execution and plain python as a normal code block, so I thought I'd borrow from that.

I see, that makes sense ... otherwise it wouldn't be possible to tell whether the code block should be rendered in Python or should be executable.

> Currently, conversion will match a codeblock that contains `{.*marimo.*}` on the first line.

Any reason to not just match Quarto and use `{python}`? Why require `marimo` to be in the tag?

>  could marimo edit notebook.md be on the tables? 

This could be on the table in the future ...! It's been requested, as you already know (#1083).

> Is there somewhere other than here to update https://docs.marimo.io/getting_started/index.html#convert-jupyter-notebooks ?

That's a fine place, but I actually just meant the docstring for now! :)

Reply:
Cool, I think I can make that change really quickly